### ISSUE TITLE
DbSeeder Throwing Exception

### ISSUE THREAD

**Issue by spboyer** [spboyer/AspNetCorePostgreSQLDockerApp#1]
Converted to RC2 getting `Method not found: 'Void Microsoft.EntityFrameworkCore.Storage.RelationalDatabaseCreator` when running `DbSeeder.SeedAsyc()`

```
public async Task SeedAsync(IServiceProvider serviceProvider)
        {
            //Based on EF team's example at https://github.com/aspnet/MusicStore/blob/dev/src/MusicStore/Models/SampleData.cs#L23
            using (var serviceScope = serviceProvider.GetRequiredService<IServiceScopeFactory>().CreateScope())
            {
                var db = serviceScope.ServiceProvider.GetService<DockerCommandsDbContext>();

                if (await db.Database.EnsureCreatedAsync())
                {
                    if (!await db.DockerCommands.AnyAsync()) {
                      await InsertSampleData(db);
                    }
                }
            }
        }
```

Appears that the issue is on ` db.Database.EnsureCreatedAsync()` when trying to create schema from migration.

```
Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.AggregateException: One or more errors occurred. (Method not found: 'Void Microsoft.EntityFrameworkCore.Storage.RelationalDatabaseCreator..ctor(Microsoft.EntityFrameworkCore.Metadata.IModel, Microsoft.EntityFrameworkCore.Storage.IRelationalConnection, Microsoft.EntityFrameworkCore.Migrations.IMigrationsModelDiffer, Microsoft.EntityFrameworkCore.Migrations.IMigrationsSqlGenerator)'.) ---> System.MissingMethodException: Method not found: 'Void Microsoft.EntityFrameworkCore.Storage.RelationalDatabaseCreator..ctor(Microsoft.EntityFrameworkCore.Metadata.IModel, Microsoft.EntityFrameworkCore.Storage.IRelationalConnection, Microsoft.EntityFrameworkCore.Migrations.IMigrationsModelDiffer, Microsoft.EntityFrameworkCore.Migrations.IMigrationsSqlGenerator)'.
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ScopedCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ScopedCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderExtensions.GetService[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.Infrastructure.AccessorExtensions.GetService[TService](IInfrastructure`1 accessor)
   at Microsoft.EntityFrameworkCore.Infrastructure.DatabaseFacade.EnsureCreatedAsync(CancellationToken cancellationToken)
   at AspNetCorePostgreSQLDockerApp.Repository.DbSeeder.<SeedAsync>d__4.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at AspNetCorePostgreSQLDockerApp.Startup.Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory, DbSeeder dbSeeder)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.AspNetCore.Hosting.Internal.ConfigureBuilder.Invoke(Object instance, IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.BuildApplication()
   at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build()
   at AspNetCorePostgreSQLDockerApp.Program.Main(String[] args)
```

@danwahlin

**Comment by spboyer** [comment]
Submitted issue: https://github.com/aspnet/EntityFramework/issues/5295

**Comment by spboyer** [comment]
@julielerman any ideas here

**Comment by julielerman** [comment]
Sorry, I've been traveling and haven't even had a chance to read the issue.Assuming closed ==resolved?

Sent from my *** Phone

On May 8, 2016, at 7:08 PM, Shayne Boyer <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #1<https://github.com/spboyer/AspNetCorePostgreSQLDockerApp/issues/1>.

-

**Comment by spboyer** [comment]
When upgrading a project and switching nuget feeds, be sure to clean the nuget packages folder. Their may be collisions with older versions hanging around.
